### DISCUSSION TITLE
Better interface for client credentials

### DISCUSSION THREAD

**Discussion by andrefurlan-db** [databricks/databricks-sdk-py#144] [Ideas]
Right now it is quite easy to use OAuthClient. It encapsulates Azure vs AWS differences and it returns a credentials_provider, which is nice. However, ClientCredentials is very crude and does nothing of the above.

One proposal [preferred] would be to encapsulate ClientCredentials within OAuthClient and have a single client to worry about.

An alternative would be to have a ClientCredentialsClient that provides a cloud-agnostic API and returns a credentials_provider.

**Comment by nfx** [comment]
the reason is that we use the same interface for Terraform and Bricks CLI - https://registry.terraform.io/providers/databricks/databricks/latest/docs#special-configurations-for-azure

all configuration properties are enumerated through [environment variables](https://registry.terraform.io/providers/databricks/databricks/latest/docs#environment-variables) in CI/CD setting. If we separate OAuth Client, this seamless enumeration experience becomes more difficult to manage, also across languages.
